substrate/frame/alliance/src/lib.rs

```diff
@@ -204,7 +204,17 @@ pub enum MemberRole {
 }
 
 /// The type of item that may be deemed unscrupulous.
-#[derive(Clone, PartialEq, Eq, RuntimeDebug, Encode, Decode, TypeInfo, MaxEncodedLen)]
+#[derive(
+	Clone,
+	PartialEq,
+	Eq,
+	RuntimeDebug,
+	Encode,
+	Decode,
+	DecodeWithMemTracking,
```

Comment:
dq, do we want/need to provide both `Decode` and `DecodeWithMemTracking`?
```suggestion
	DecodeWithMemTracking,
```

I see that `pub trait DecodeWithMemTracking: Decode {}`

Reply:
`DecodeWithMemTracking` requires also for the type to implement `Decode`. But if we derive only `DecodeWithMemTracking`, as far as I know `Decode` will not be derived automatically. So we need to specify both.

Reply:
aha, ok, ok, thank you :)